Renata/package setup

r-akemii opened the pull request:
* Creates functions for importing network, set population source, importing country subdivisions, importing population pyramid and import buildings (from both OSM and Bing) and points of interest (amenities).

pedrocamargo committed:
Initial setup of model-building package

pedrocamargo committed:
some data_changes

r-akemii committed:
Changes in population

r-akemii committed:
Changes in overwrite

r-akemii committed:
Correct model and functions

r-akemii committed:
Fix the code

Implemented corrections in set_population_source and import_population.

r-akemii committed:
Update files

r-akemii committed:
Create verification triggers

r-akemii committed:
Update model

r-akemii committed:
Update model

r-akemii committed:
Create population pyramid

r-akemii committed:
Update model

r-akemii committed:
Import amenities

r-akemii committed:
Building import and updates

r-akemii committed:
Update model

r-akemii committed:
Update model

Correct bugs in amenity and building imports

r-akemii committed:
Fix functions

r-akemii committed:
Update model.py

pedrocamargo commented on gpbp/data/population_file_address.py:
```diff
@@ -0,0 +1,53 @@
+        except AttributeError:
+            country_iso = pycountry.countries.get(common_name=model_place).alpha_3.lower()
+
+        if ' ' in model_place:
```
Nao precisa do if, e o "else, pass" tb nao faz nada

pedrocamargo commented on gpbp/data/population_file_address.py:
```diff
@@ -0,0 +1,53 @@
+        else:
+            pass
+
+        base_url = 'https://data.humdata.org'
```
Precisamos de um arquivo de parametetros para colocar essas coisas todas. Hard coding é um negócio que quebra muito fácil

pedrocamargo commented on gpbp/data/population_raster.py:
```diff
@@ -8,9 +8,9 @@
 from scipy.sparse import coo_matrix
 
-from functions.country_main_area import get_main_area
+from notebooks.functions.country_main_area import get_main_area
```
Notebooks não faz parte do pacote, então as funções não podem estar lá, mas dentro do proprio gbpb

pedrocamargo commented on gpbp/data_retrieval/osm_building_import.py:
```diff
@@ -0,0 +1,61 @@
+            pass
+
+        single_tuple_1 = (int(row.id), idx[1])
+        single_tuple_2 = (int(row.area), idx[1])
```
Area é inteiro mesmo?

pedrocamargo commented on gpbp/data_retrieval/osm_building_import.py:
```diff
@@ -0,0 +1,61 @@
+        single_tuple_1 = (int(row.id), idx[1])
+        single_tuple_2 = (int(row.area), idx[1])
+
+        qry = f'UPDATE zones SET {field_name_1}=? WHERE zone_id=?;'
```
Esses updates demoram demais.  O negocio é acumular todos os parametros numa lista de listas e usar **executemany** depois do loop

pedrocamargo commented on gpbp/data_retrieval/osm_tags/area_microsoft_buildings.py:
```diff
@@ -0,0 +1,27 @@
+    zoning = project.zoning
+    fields = zoning.fields
+
+    try:
```
Try/except não pode ser usado para workflow. O que precisamos é verificar se o campo existe e, se não, adicioná-lo

pedrocamargo commented on gpbp/data_retrieval/osm_tags/area_microsoft_buildings.py:
```diff
@@ -0,0 +1,27 @@
+    
+    list_of_tuples = list(zip(total_area.values(), total_area.keys()))
+
+    qry = f'UPDATE zones SET microsoft_building_area=? WHERE zone_id=?;'
```
o **f'** não faz sentido aqui, pois vc nao está formatando esse string

pedrocamargo commented on gpbp/data_retrieval/osm_tags/count_microsoft_buildings.py:
```diff
@@ -0,0 +1,28 @@
+    zoning = project.zoning
+    fields = zoning.fields
+
+    try:
```
de novo

pedrocamargo replied:
O proprio AequilibraE deve ter um metoido pra listar os campos que jah existem

pedrocamargo commented on gpbp/data_retrieval/osm_tags/count_microsoft_buildings.py:
```diff
@@ -0,0 +1,28 @@
+
+    list_of_tuples = list(zip(count_buildings.values(), count_buildings.keys()))
+
+    qry = f'UPDATE zones SET microsoft_building_count=? WHERE zone_id=?;'
```
**f'** de novo

pedrocamargo commented on gpbp/data_retrieval/osm_tags/count_microsoft_buildings.py:
```diff
@@ -0,0 +1,28 @@
+
+    count_buildings = buildings_by_zone.groupby('zone_id').count()['area'].to_dict()
+
+    for key in missing_idx:
```
Não faz muito sentido adicionar zeros no dicionário pra depois fazer um executemany.  Melhor mudar pra dois update tables, o primeiro colocando tudo como zero, e o segundo dando update nos que sao diferentes de zero, o que elimina essa parte do codigo

pedrocamargo commented on gpbp/data_retrieval/osm_tags/import_osm_buildings.py:
```diff
@@ -0,0 +1,53 @@
+from gpbp.data_retrieval.osm_tags.osm_tag_values import building_values
+
+
+def import_osm_buildings(model_place:str, project:Project, osm_data:dict):
```
o osm_buildings é MUITO semelhante ao osm_amenities..  Me dá a impressão que deve ter um jeito de simplificar o código de forma ter uma função genérica que faça os 2

pedrocamargo commented on gpbp/data_retrieval/osm_tags/set_bounding_boxes.py:
```diff
@@ -0,0 +1,35 @@
+import geopandas as gpd
+import pandas as pd
+from math import ceil, sqrt
+from notebooks.functions.country_borders import get_country_borders
```
de novo

pedrocamargo commented on gpbp/model.py:
```diff
@@ -0,0 +1,136 @@
+
+class Model:
+    def __init__(self, network_path: str, model_place: str = None):
+        # TODO: MAKE SURE THE CLASS WORKS FOR USING THE MODEL AND CREATING IT
```
Esse TODO ainda é um TODO? Ou pode ir embora?

pedrocamargo commented on gpbp/model_creation/country_borders.py:
```diff
@@ -0,0 +1,23 @@
+import sqlite3
+
+
+def get_country_borders(country_name: str):
```
Acho que podemos mudar TODO o acesso a areas de paises e suas subdivisoes para esse cara aqui:

https://www.geoboundaries.org/api.html

pedrocamargo commented on gpbp/model_creation/country_subdivisions.py:
```diff
@@ -0,0 +1,21 @@
+
+def get_subdivisions(country_name:str, subdivisions:int):
+    
+    url = 'https://github.com/pedrocamargo/road_analytics/releases/download/v0.1/subdivisions.gpkg'
```
Ver comentario sobre country_borders

pedrocamargo commented:
Rodar o flake8 vai ajudar a revelar um monte dos problemas que eu comentei

r-akemii committed:
Update country subdivision and population source

Make changes in country subdivision imports, and subsequently in the population sources files.

r-akemii committed:
Changes in amenities, buildings and RAI

r-akemii committed:
Fix control flow and data retrievals

r-akemii committed:
Fix OSM data import

r-akemii committed:
Fix Population Import

r-akemii committed:
Remove files

r-akemii committed:
Fix raw_population import

r-akemii committed:
Fix function names
